src/instructlab/model/backends/common.py

```diff
@@ -37,23 +61,117 @@ class ServerException(Exception):
     """An exception raised when serving the API."""
 
 
-def get_model_template(
-    model_family: str, model_path: pathlib.Path
+def get_in_memory_model_template(
+    model_family: str, model_arch: str
 ) -> Tuple[str, str, str]:
-    eos_token = "<|endoftext|>"
-    bos_token = ""
-    template = ""
-    resolved_family = get_model_family(model_family, model_path)
+    """
+    Retrieve an in-memory chat template based on the model family and architecture
+
+    args
+        model_family (str): family that the model belongs to
+        model_arch (str): architecture of the model
+
+    returns
+        template (str): resolved chat template
+        bos_token (str): Beginning of sentence token
+        eos_token (str): End of sentence token
+    """
+    template = eos_token = bos_token = ""
+
     logger.debug(
-        "Searching hard coded model templates for model family %s's template",
-        resolved_family,
+        "Searching in-memory model templates for model family %s and arch %s",
+        model_family,
+        model_arch,
     )
+
+    # Try to match both model and arch
     for template_dict in templates:
-        if template_dict["model"] == resolved_family:
+        if (
+            template_dict.get("family") == model_family
+            and template_dict.get("arch") == model_arch
+        ):
             template = template_dict["template"]
-            if template_dict["model"] == "mixtral":
-                eos_token = "</s>"
-                bos_token = "<s>"
+            bos_token = template_dict["special_tokens"].bos.token
+            eos_token = template_dict["special_tokens"].eos.token
+
+    # If no match, try matching on model only
+    if template == "":
+        for template_dict in templates:
+            if template_dict.get("family") == model_family:
+                template = template_dict["template"]
+                bos_token = template_dict["special_tokens"].bos.token
+                eos_token = template_dict["special_tokens"].eos.token
+
+    logger.debug("no match found for supplied model family and architecture")
+    return template, eos_token, bos_token
+
+
+def format_template(template: str, bos_token: str, eos_token: str) -> str:
```

Comment:
can we add docstrings for all of these new functions please?

Reply:
yes will do

Reply:
updated, thanks